src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
-}
 
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const keyRemove of action.keysToRemove) {
+          delete state[keyRemove];
+        }
+
+        break;
+
+      case 'clear':
+        for (const keyClear in state) {
+          delete state[keyClear];
+        }
+
+        break;
+    }
```

Comment:
[checklist](https://github.com/mate-academy/js_stateful-object/blob/master/checklist.md) 4th item, add default case for error handling